brains/andalite.py

```diff
@@ -0,0 +1,107 @@
+import ast
+import asyncio
+import concurrent
+import json
+import pprint
+import queue
+import threading
+import websockets
+
+from sapiens import Sapiens
+
+class Andalite(Sapiens):
+    name = 'andalite'
+    lastSent = {}
+
+    def __init__(self, game, args=None):
+        Sapiens.__init__(self, game, args)
+        self.telepathy = Telepathy()
+
+    def Step(self):
+        self.sendMemory()
+        self.receiveBytes()
+        return Sapiens.Step(self)
+
+    def Event(self, evt):
+        Sapiens.Event(self, evt)
+
+    def Path(self):
+        return Sapiens.Path(self)
+
+    def Victory(self):
+        return Sapiens.Victory(self)
+
+    def sendMemory(self):
+        self.sendNewBytes(0xcc24, 11) # menu data
+
+    def sendNewBytes(self, start, lengthBytes):
+        data = {'offset': start, 'data': self.game.PeekMemoryRegion(start, lengthBytes)} # menu data
+        lastSentData = self.lastSent.get(start)
+        if lastSentData != data:
+            self.telepathy.outboundMemoryReads.put_nowait(str(data))
+            self.lastSent[start] = data
+
+    def receiveBytes(self):
+        try:
+            data = self.telepathy.inboundMemoryWrites.get_nowait()
+            start = data.get('offset')
+            lastData = self.lastSent.get(start)
+
+            if lastData == None or start != None and lastData.get('data') != data.get('data'):
+                print("writing data")
+                self.game.PokeMemoryRegion(data['offset'], data['data'])
+                self.lastSent['offset'] = data
+        except queue.Empty:
+            pass
+
+class Telepathy:
+    inboundMemoryWrites = queue.Queue()
+    outboundMemoryReads = queue.Queue()
+
+    def __init__(self):
+        thread = threading.Thread(target=self.threadMain, args=())
+        thread.daemon = True
+        thread.start()
+        #asyncio.get_event_loop().run_forever()
+    def threadMain(self):
```

Comment:
(PEP8) style nit: i think this method needs one more `\n` before it and one fewer `\n` after it

Reply:
also it looks like all the new methods are lowerCamel while the existing stuff is UpperCamel and the PEP8 standard (as well as that of the libretro C API) is snake_case.  we should probably standardize on one style... can be decided later and done in a different commit tho

Reply:
agreed. let's snap to snake_case since that's more pythonic

Reply:
created issue #4